minfraud/validation.py

```diff
@@ -379,14 +379,38 @@ def _uuid(s: str) -> str:
     raise ValueError
 
 
-validate_report = Schema(
+def _transaction_id(s: Optional[str]) -> str:
+    if isinstance(s, str) and len(s) > 0:
+        return s
+    raise ValueError
+
+
+_validate_report_schema = Schema(
     {
         "chargeback_code": str,
-        Required("ip_address"): _ip_address,
+        "ip_address": _ip_address,
         "maxmind_id": _maxmind_id,
         "minfraud_id": _uuid,
         "notes": str,
         Required("tag"): _tag,
-        "transaction_id": str,
+        "transaction_id": _transaction_id,
     },
 )
+
+
+def _validate_at_least_one_identifier_field(report):
+    optional_fields = ["ip_address", "maxmind_id", "minfraud_id", "transaction_id"]
+    if not any(field in report for field in optional_fields):
```

Comment:
This is probably bikeshedding, but expressing it as a set operation seems a bit simpler to me:

```suggestion
    optional_fields = {"ip_address", "maxmind_id", "minfraud_id", "transaction_id"}
    if not optional_fields & report.keys():
```

Reply:
That said, maybe we should handle the empty string for these too like we do for the other APIs.

Reply:
The individual field validation that exists should already disallow empty strings for all of these fields. The only one we might need to check is minfraid_id to disallow the zero UUID